### PR TITLE
Add scoring highlighter plugin

### PR DESCRIPTION
**Pull request by AnastasiaArcadia**
# Pull request summary

Related to [MS-447](https://oat-sa.atlassian.net/browse/MS-447).

Add scorer's Highlight plugin

# How to test

On MS side:

-   Login as a scorer
-   Go to the task page
-   Click on the highlighter icon button
-   Select some text, click on the color button - text should be highlighted
-   Click on the eraser button - highlights should be cleared

### CODE REVIEW

**Review by anshul119 — APPROVED** [review]
Works well!

File: views/js/previewer/plugins/content/scss/highlighterTray.scss [hunk 1]
```diff
@@ -0,0 +1,80 @@
+.highlighter-container {
+    $highlighterBackground: #1F1F1F;
```
Comment (ampaveliev) [inline views/js/previewer/plugins/content/scss/highlighterTray.scss:2]: Mixing variables declarations always look messy. Does it really necessary? Also bear in mind  that  local **CSS** variables can't be used in ie11

File: views/js/previewer/plugins/content/scoringHighlighter.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    return pluginFactory({
+        name: 'highlighter',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init() {
+            const testRunner = this.getTestRunner();
+
+            if (!window.getSelection) {
```
Comment (ampaveliev) [inline views/js/previewer/plugins/content/scoringHighlighter.js]: Is it necessary? Seems it won't be used in ie8. And be more careful with throwing errors:
1. Make sure that throwing error doesn't break the whole app
2. JSDoc throwing error

File: views/js/previewer/plugins/content/scoringHighlighter.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                throw new Error('Browser does not support getSelection()');
+            }
+            this.selection = window.getSelection();
+
+            this.eventListener = e => {
+                if (e.data.event === 'setIndex') {
+                    // Applying any highlighIndex received from parent
+                    this.highlighter.highlightFromIndex(e.data.payload);
+                } else if (this.$highlighterTray) {
+                    if (e.data.event === 'hide') {
```
Comment (ampaveliev) [inline views/js/previewer/plugins/content/scoringHighlighter.js:70]: We can avoid nesting if operators usage

File: views/js/controller/ItemResultPreviewer/itemResultPreviewer.js [hunk 4]
```diff
@@ -24,7 +32,8 @@ define(['taoItems/previewer/factory'], function(previewerFactory) {
             previewerFactory(type, uri, state, {
                 view: 'scorer',
                 fullPage: true,
-                hideActionBars: true
+                hideActionBars: true,
+                plugins
```
Comment (ampaveliev) [inline views/js/controller/ItemResultPreviewer/itemResultPreviewer.js:36]: Does it affect other apps except manual scoring someway? Or is it ok? 🤔

**Review by ampaveliev — COMMENTED** [review]
Just code review. Looks fine, but some comments should be resolved

File: views/js/controller/ItemResultPreviewer/itemResultPreviewer.js (hunk 4 above)
Comment (anshul119) [reply views/js/controller/ItemResultPreviewer/itemResultPreviewer.js:36]: no, it doesn't ;) that's the good thing about lti outcome ui, only MS uses it

File: views/js/previewer/plugins/content/scoringHighlighter.js (hunk 3 above)
Comment (AnastasiaArcadia) [reply views/js/previewer/plugins/content/scoringHighlighter.js:70]: Nesting is not so big

**Review by ampaveliev — APPROVED** [review]

**Pull request merged**